internal/internal_event_handlers.go

```diff
@@ -153,7 +153,7 @@ type (
 		cancelHandler   func()                                                                     // A cancel handler to be invoked on a cancel notification
 		signalHandler   func(name string, input *commonpb.Payloads, header *commonpb.Header) error // A signal handler to be invoked on a signal event
 		queryHandler    func(queryType string, queryArgs *commonpb.Payloads, header *commonpb.Header) (*commonpb.Payloads, error)
-		updateHandler   func(name string, args *commonpb.Payloads, header *commonpb.Header, callbacks UpdateCallbacks)
+		updateHandler   func(name string, ID string, args *commonpb.Payloads, header *commonpb.Header, callbacks UpdateCallbacks)
```

Comment:
```suggestion
		updateHandler   func(name string, id string, args *commonpb.Payloads, header *commonpb.Header, callbacks UpdateCallbacks)
```

And elsewhere. Parameters should not be capitalized.